run/run.py

```diff
@@ -254,6 +244,29 @@ def main(platform_id, platform, args, config):
     print('Response text:', response.text)
 
 
+def setup_wpt(mainargs, platform, config, logger):
+    wpt_setup_commands = [
+        ['git', 'reset', '--hard', 'HEAD'],  # For wpt.patch
+        ['git', 'checkout', 'master'],
+        ['git', 'pull'],
+        ['./wpt', 'manifest', '--work'],
+    ]
+    for command in wpt_setup_commands:
+        return_code = subprocess.check_call(command, cwd=config['wpt_path'])
+        assert return_code == 0, (
+            'Got non-0 return code: '
+            '%d from command %s' % (return_code, command))
+
+    patch_wpt(config, platform)
+
+    if mainargs.wpt_sha:
+        return mainargs.wpt_sha
+    else:
+        sha_finder = shas.SHAFinder(logger)
```

Comment:
Do you think that this code path should later be retired by always requiring the sha to be provided? We'll end up here if there's a typo on the command line, so maybe emitting a warning that we picked a commit would make sense. Or a TODO to retire this stuff.

Reply:
I'll take the "TODO" path for now, so I/we can think more about this